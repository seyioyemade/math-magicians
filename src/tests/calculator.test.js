import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../components/calculator';

describe('Calculator', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><Calculator /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
