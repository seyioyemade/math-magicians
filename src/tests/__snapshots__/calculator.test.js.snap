// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
Array [
  <header>
    <h1>
      Math Magicians
    </h1>
    <nav>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="calculator-page"
  >
    <h2>
      Let's do some math!
    </h2>
    <section>
      <table>
        <tr>
          <th
            colSpan="4"
          >
            0
          </th>
        </tr>
        <tr>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              AC
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              +/-
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              %
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              รท
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              7
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              8
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              9
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              x
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              4
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              5
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              6
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              -
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              1
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              2
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              3
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              +
            </button>
          </td>
        </tr>
        <tr>
          <td
            colSpan="2"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              0
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              .
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              =
            </button>
          </td>
        </tr>
      </table>
    </section>
  </div>,
]
`;
